src/gui/editors/AutomationEditor.cpp

```diff
@@ -2282,6 +2285,35 @@ const AutomationPattern* AutomationEditorWindow::currentPattern()
 	return m_editor->currentPattern();
 }
 
+void AutomationEditorWindow::dropEvent(QDropEvent *_de)
+{
+	QString type = StringPairDrag::decodeKey( _de );
+	QString val = StringPairDrag::decodeValue( _de );
+	if( type == "automatable_model" )
+	{
+		AutomatableModel * mod = dynamic_cast<AutomatableModel *>(
+				Engine::projectJournal()->
+					journallingObject( val.toInt() ) );
+		if( mod != NULL )
+		{
+			if( m_editor->m_pattern->firstObject() )
+			{
+				m_editor->m_pattern->objectDestroyed( m_editor->m_pattern->firstObject()->id() );
+			}
+			m_editor->m_pattern->clear();
```

Comment:
@diizy do we normally clear the pattern when we drag in a new item?  Currently we can link one automation event to multiple similar controls, right?